libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -663,9 +641,7 @@ - (void)finalizeAuthCompletion
     // Assign the identity data to the current user
     NSAssert([SFUserAccountManager sharedInstance].currentUser != nil, @"Current user should not be nil at this point.");
     [[SFUserAccountManager sharedInstance] applyIdData:self.idCoordinator.idData];
-
-    // Save the accounts
-    [[SFUserAccountManager sharedInstance] saveAccounts:nil];
+    [[SFUserAccountManager sharedInstance] updateAccount:[SFUserAccountManager sharedInstance].currentUser];
```

Comment:
I think this is probably the crux of what's bothering me—and the more so given the red flag that we're assuming `currentUser` is the one being authenticated at any given time.

I would prefer a somewhat different approach to how we're refactoring authentication in this class.  I'm okay with the idea that we pass in credentials to `login*` as opposed to users—it's a reasonable decoupling at that stage.  But:

1. I don't think we need to preemptively tie a set of credentials to a user account before we authenticate in this class.
2. Rather than making assumptions at the end of this flow about what user account is associated with the updated credentials, I would prefer that:

    1. We "look up" any user account associated with the UserID/OrgID compound unique key that will be associated with the credentials, and either apply the updates to that user, or create a new user account with those credentials, if one didn't previously exist.
    2. Return that user in the completion blocks of this class (and probably in the `kSFAuthenticationManagerFinishedNotification` note, though at some point I'd like to clean up all of these post-auth reports, but that's another day).

This approach allows us to:

1. Pivot away from trying to do up-front staging of an association between credentials and a user—the edge cases of "empty" credentials and users are actually the reason we have a lot of IMO tenuous decoupled identities in `SFUserAccountManager`, that could simply go away.
2. Gets `SFAuthenticationManager` wholly out of the business of deciding what the _current_ user context is—let's move that responsibility onto the caller—making it almost entirely user-agnostic.
3. Probably allows us to minimize refactoring of `SFOAuthCredentials`, which is going to make me sleep better at night.